lib/block-supports/typography.php

```diff
@@ -87,67 +91,68 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 		}
 	}
 
-	// Font Family.
-	if ( $has_font_family_support ) {
-		$has_font_family = isset( $block_attributes['style']['typography']['fontFamily'] );
-		// Apply required class and style.
-		if ( $has_font_family ) {
-			$font_family = $block_attributes['style']['typography']['fontFamily'];
-			if ( strpos( $font_family, 'var:preset|font-family' ) !== false ) {
-				// Get the name from the string and add proper styles.
-				$index_to_splice  = strrpos( $font_family, '|' ) + 1;
-				$font_family_name = substr( $font_family, $index_to_splice );
-				$styles[]         = sprintf( 'font-family: var(--wp--preset--font-family--%s);', $font_family_name );
-			} else {
-				$styles[] = sprintf( 'font-family: %s;', $block_attributes['style']['color']['fontFamily'] );
+	if ( ! $skip_typography_serialization ) {
```

Comment:
What if we added this check to each of the existing (has_font_family_support, has_font_style_support, etc) instead? There're up to 4 levels of conditionals in some places. It seems like the code paths would be clearer if we unnest that a bit.

Reply:
Of the options discussed, I think my preference would be:

1. Add check to every individual property.
2. Try bail early.
3. Nesting.

For consideration: if/when all typographic properties are consolidated under typography all of them will have the same key, _at that point_, bailing early will make more sense.

Reply:
> Add check to every individual property.

Done in 65238e8a5f 🙂